browser/ui/views/split_view/split_view_location_bar.cc

```diff
@@ -0,0 +1,342 @@
+              .AddChild(views::Builder<views::ImageView>()
+                            .CopyAddressTo(&safety_icon_)
+                            .SetImage(ui::ImageModel::FromVectorIcon(
+                                kLeoWarningTriangleOutlineIcon,
+                                kColorBraveSplitViewMenuItemIcon, 14)))
+              .AddChild(
+                  views::Builder<views::Label>()
+                      .SetText(u"https")
+                      .CopyAddressTo(&https_with_strike_)
+                      .SetEnabledColorId(kColorOmniboxSecurityChipDangerous))
+              .AddChild(views::Builder<views::Label>()
+                            .SetText(u"://")
+                            .CopyAddressTo(&scheme_separator_)
+                            .SetEnabledColorId(kColorBraveSplitViewUrl))
+              .AddChild(views::Builder<views::Label>()
+                            .CopyAddressTo(&url_)
+                            .SetEnabledColorId(kColorBraveSplitViewUrl)
+                            .SetElideBehavior(gfx::ElideBehavior::ELIDE_HEAD)))
+      .BuildChildren();
+
+  for (auto url_part : {https_with_strike_, scheme_separator_, url_}) {
+    // Adjust font size.
+    constexpr auto kURLSize = 12;
+    url_part->SetFontList(url_part->font_list().DeriveWithSizeDelta(
+        kURLSize - url_part->font_list().GetFontSize()));
+  }
+
+  for (auto scheme_part : {https_with_strike_, scheme_separator_}) {
+    // Remove child spacing
+    scheme_part->SetProperty(views::kMarginsKey,
+                             gfx::Insets().set_right(-kChildSpacing));
+  }
+
+  // Strike through the "https"
+  https_with_strike_->SetFontList(
+      https_with_strike_->font_list().DeriveWithStyle(
+          gfx::Font::FontStyle::STRIKE_THROUGH));
+
+  if (!prefs_) {
+    CHECK_IS_TEST();
+    return;
+  }
+
+  prevent_url_elision_.Init(
+      omnibox::kPreventUrlElisionsInOmnibox, prefs_,
+      base::BindRepeating(&SplitViewLocationBar::UpdateURLAndIcon,
+                          base::Unretained(this)));
+}
+
+SplitViewLocationBar::~SplitViewLocationBar() = default;
+
+// static
+SplitViewLocationBar* SplitViewLocationBar::Create(PrefService* service,
+                                                   views::View* web_view) {
+  CHECK(web_view->GetWidget());
+  auto* location_bar = new SplitViewLocationBar(service);
+
+  views::Widget::InitParams params(views::Widget::InitParams::TYPE_CONTROL);
+  params.parent = web_view->GetWidget()->GetNativeView();
+  params.activatable = views::Widget::InitParams::Activatable::kNo;
+  params.delegate = location_bar;
+  auto* widget = new views::Widget();
+  widget->Init(std::move(params));
+
+  location_bar->view_observation_.Observe(web_view);
+  location_bar->UpdateVisibility();
+  location_bar->UpdateBounds();
+
+  widget->Show();
+
+  return location_bar;
+}
+
+void SplitViewLocationBar::SetWebContents(content::WebContents* web_contents) {
+  if (this->web_contents() == web_contents) {
```

Comment:
use of `this` is not necessary

Reply:
Actually, it's needed as |web_contents| is also parameter name here.

Reply:
then you should change the param name

Reply:
Changed